### DISCUSSION TITLE
Load referenced DataPortal object in blazor

### DISCUSSION THREAD

**Discussion by Prashant3110kk** [MarimerLLC/csla#2299] [Questions] [answered]
I have a DataPortal object called AgencyList I'm loading AgencyList as below
`        protected override async Task OnInitializedAsync()
        {
            await vm.RefreshAsync(() => Csla.DataPortal.FetchAsync<AgencyList>());
        }`
problem here is AgencyList Contains fewer data(i.e. agencyID, name), I have one more DataPortal object called AgencyEdit(agencyID) which got all data I needed, is there ant way call AgencyEdit(agencyID) from front end? without making any server side changes?

Any help is appreciated, Thanks

**Comment by rockfordlhotka** [comment, marked as answer]
The idea behind CSLA is that you design your object model to meet the needs of each user scenario. This is often driven by the requirements of the UI.

My recommendation is that if you have a read-only list you are displaying to the user, that you update the list to contain the fields (properties) required by the scenario and UI.

However, if you are retrieving a single object to display a lot of properties to the user, _and also_ you have a page that edits those very same properties, then I'd personally re-use the same read-write business type (in your case AgencyEdit) for both pages.

I would _not_ create a read-only list based on AgencyEdit objects though, because that'll often cause major performance issues over time. You'd be misusing AgencyEdit (giving it multiple responsibilities, and so violating the single responsibility pattern). That might get your app working _today_, but you'll curse yourself (or someone will curse you) in the future.
